modules/drivers/druid-jdbc/src/metabase/driver/druid_jdbc.clj

```diff
@@ -0,0 +1,183 @@
+(ns metabase.driver.druid-jdbc
+  (:require
+   [cheshire.core :as json]
+   [clj-http.client :as http]
+   [clojure.string :as str]
+   [clojure.walk :as walk]
+   [java-time.api :as t]
+   [metabase.driver :as driver]
+   [metabase.driver.common :as driver.common]
+   [metabase.driver.sql-jdbc.connection :as sql-jdbc.conn]
+   [metabase.driver.sql-jdbc.execute :as sql-jdbc.execute]
+   [metabase.driver.sql-jdbc.sync :as sql-jdbc.sync]
+   [metabase.driver.sql.query-processor :as sql.qp]
+   [metabase.driver.sql.query-processor.util :as sql.qp.u]
+   [metabase.driver.sql.util.unprepare :as unprepare]
+   [metabase.lib.field :as lib.field]
+   [metabase.lib.metadata :as lib.metadata]
+   [metabase.query-processor.store :as qp.store]
+   [metabase.query-processor.util.add-alias-info :as add]
+   [metabase.util.honey-sql-2 :as h2x]
+   [metabase.util.log :as log])
+  (:import
+   (java.sql ResultSet Types)
+   (java.time ZonedDateTime)))
+
+(set! *warn-on-reflection* true)
+
+(driver/register! :druid-jdbc :parent :sql-jdbc)
+
+(doseq [[feature supported?] {:set-timezone            true
+                              :expression-aggregations true}]
+  (defmethod driver/database-supports? [:druid-jdbc feature] [_driver _feature _db] supported?))
+
+(defmethod sql-jdbc.conn/connection-details->spec :druid-jdbc
+  [_driver {:keys [host port] :as _db-details}]
+  (merge {:classname   "org.apache.calcite.avatica.remote.Driver"
+          :subprotocol "avatica:remote"
+          :subname     (str "url=" host ":" port "/druid/v2/sql/avatica/;transparent_reconnection=true")}
+         (when (some? (driver/report-timezone))
+           {:sqlTimeZone (driver/report-timezone)})))
+
+(defmethod driver/db-default-timezone :druid-jdbc
+  [_driver _database]
+  "UTC")
+
+(defmethod driver/db-start-of-week :druid-jdbc
+  [_]
```

Comment:
```suggestion
  [_driver]
```

A lot of the older code has `_` everywhere but it's always good to do stuff like `_driver` in new code, it's way clearer.

Reply:
Thanks, I'll keep that in mind. I've updated param names also in other places in the driver.